apps/api/src/app/events/e2e/trigger-multicast.e2e.ts

```diff
@@ -0,0 +1,573 @@
+        { type: TriggerRecipientsTypeEnum.TOPIC, topicKey: TOPIC_KEY_PREFIX + '1' },
+      ];
+
+      // second topic: subscribers B, D
+      const secondTopic: TriggerRecipients = [
+        { type: TriggerRecipientsTypeEnum.TOPIC, topicKey: TOPIC_KEY_PREFIX + '2' },
+      ];
+
+      /*
+       * add subscribers to second topic from first topic ,
+       * now second topic subscribers are not duplicated with first topic subscribers (A, B, C, D)
+       */
+      await addSubscribersToTopic(session, bootstrapFirstTopic, firstTopicSubscribers);
+
+      /*
+       * in this case we send the same subscriber twice,
+       * but the newSubscriber that is not duplicated and should be sent only once by topic
+       * singleSubscribers contains: A, B
+       * firstTopic contains: A, B
+       */
+      const command: TriggerMulticastCommand = buildTriggerMulticastCommandMock({
+        to: [...firstTopic, ...secondTopic],
+        organizationId: session.organization._id,
+        environmentId: session.environment._id,
+        userId: session.user._id,
+      }) as any;
+
+      await triggerMulticast.execute(command);
+
+      expect(sendToProcessSubscriberServiceStub.called).to.be.true;
+
+      const singleSubscribersDefine = mapSubscriberToSubscriberDefine(firstTopicSubscribers);
+
+      const firstCallStubArgs = sendToProcessSubscriberServiceStub.getCall(0).args;
+      expect(firstCallStubArgs[0]).to.deep.equal(command);
+      expect(firstCallStubArgs[1]).to.include.deep.members([]);
+      expect(firstCallStubArgs[2]).to.be.equal(SubscriberSourceEnum.SINGLE);
+
+      // we check by second topic subscribers without duplication
+      const secondTopicSubscriberDefine = mapSubscriberToSubscriberDefine(secondTopicSubscribers);
+
+      const secondCallStubArgs = sendToProcessSubscriberServiceStub.getCall(1).args;
+      expect(secondCallStubArgs[0]).to.deep.equal(command);
+      expect(secondCallStubArgs[1]).to.include.deep.members(secondTopicSubscriberDefine);
+      expect(secondCallStubArgs[2]).to.be.equal(SubscriberSourceEnum.TOPIC);
+    });
+
+    it('should deduplicate subscribers across single subscribers and topics', async () => {
+      const newSubscriber = await subscriberService.createSubscriber();
+      await addSubscribersToTopic(session, bootstrapFirstTopic, [newSubscriber]);
+
+      // first topic: subscribers A, B
+      const firstTopic: TriggerRecipients = [
+        { type: TriggerRecipientsTypeEnum.TOPIC, topicKey: TOPIC_KEY_PREFIX + '1' },
+      ];
+
+      // second topic: subscribers B, D
+      const secondTopic: TriggerRecipients = [
+        { type: TriggerRecipientsTypeEnum.TOPIC, topicKey: TOPIC_KEY_PREFIX + '2' },
+      ];
+
+      /*
+       * add subscribers to second topic from first topic ,
+       * now second topic subscribers are not duplicated with first topic subscribers (A, B, C, D)
+       */
+      await addSubscribersToTopic(session, bootstrapFirstTopic, firstTopicSubscribers);
```

Comment:
It is confusing because it is wrong.. let me fix it.
While fixing it i noticed an additional mistake which was to test by `to.include.deep.members` and not `to.have.deep.members` :/